keras_core/backend/jax/layer.py

```diff
@@ -1,4 +1,18 @@
+from keras_core.backend.common import distribute_scope
+
+
 class JaxLayer:
     def _post_build(self):
         """Can be overriden to perform post-build actions."""
-        pass
+        if not distribute_scope.in_distribute_scope():
+            return
+
+        distribution = distribute_scope.get_distribute_scope().distribute
```

Comment:
We could do this at any point after `build()`, right? Including at the start of `fit()`. This means that a `compile()` argument is viable.

Reply:
In the LLM case, doing it after the init of whole model will be too late, since the overall weights will OOM the single device. We should shard the weights while we build the model.